### PR TITLE
Scripts/Commands: Convert argument parsing of event commands to new system

### PR DESCRIPTION
**Pull request by Carbenium**
**Changes proposed:**

-  Scripts/Commands: Convert argument parsing of event commands to new system

**Target branch(es):** 3.3.5/master

- [x] 3.3.5
- [ ] master

### CODE REVIEW

File: src/server/scripts/Commands/cs_event.cpp [hunk 1]
```diff
@@ -80,28 +81,18 @@ class event_commandscript : public CommandScript
...
-        if (eventId >= events.size())
+        if (*eventId >= events.size())
         {
             handler->SendSysMessage(LANG_EVENT_NOT_EXIST);
             handler->SetSentErrorMessage(true);
             return false;
         }
 
-        GameEventData const& eventData = events[eventId];
+        GameEventData const& eventData = events[*eventId];
```
Comment (Treeston) [inline src/server/scripts/Commands/cs_event.cpp:95]: is implicit conversion not sufficient here?

File: src/server/scripts/Commands/cs_event.cpp [hunk 2]
```diff
@@ -80,28 +81,18 @@ class event_commandscript : public CommandScript
...
-        char* id =  handler->extractKeyFromLink((char*)args, "Hgameevent");
-        if (!id)
-            return false;
-
-        uint32 eventId = atoul(id);
-
         GameEventMgr::GameEventDataMap const& events = sGameEventMgr->GetEventMap();
 
-        if (eventId >= events.size())
+        if (*eventId >= events.size())
```
Comment (Treeston) [inline src/server/scripts/Commands/cs_event.cpp:88]: same question

File: src/server/scripts/Commands/cs_event.cpp [hunk 3]
```diff
@@ -110,13 +101,13 @@ class event_commandscript : public CommandScript
         }
 
         GameEventMgr::ActiveEvents const& activeEvents = sGameEventMgr->GetActiveEventList();
-        bool active = activeEvents.find(eventId) != activeEvents.end();
+        bool active = activeEvents.find(*eventId) != activeEvents.end();
```
Comment (Treeston) [inline src/server/scripts/Commands/cs_event.cpp:104]: yeah like same question for all of these, `eventId` implicitly converts to `uint32`

File: src/server/scripts/Commands/cs_event.cpp (hunk 1 above)
Comment (Carbenium) [reply src/server/scripts/Commands/cs_event.cpp:95]: `../src/server/scripts/Commands/cs_event.cpp:95:49: error: invalid user-defined conversion from ‘const Trinity::ChatCommands::Variant<Trinity::ChatCommands::Hyperlink<Trinity::Hyperlinks::LinkTags::gameevent>, unsigned int>’ to ‘std::vector<GameEventData>::size_type’ {aka ‘long unsigned int’} [-fpermissive]
`

File: src/server/scripts/Commands/cs_event.cpp (hunk 2 above)
Comment (Carbenium) [reply src/server/scripts/Commands/cs_event.cpp:88]:
```
../src/server/scripts/Commands/cs_event.cpp:88:21: error: no match for ‘operator>=’ (operand types are ‘const Trinity::ChatCommands::Variant<Trinity::ChatCommands::Hyperlink<Trinity::Hyperlinks::LinkTags::gameevent>, unsigned int>’ and ‘std::vector<GameEventData>::size_type’ {aka ‘long unsigned int’})
```

File: src/server/scripts/Commands/cs_event.cpp (hunk 3 above)
Comment (Carbenium) [reply src/server/scripts/Commands/cs_event.cpp:104]:
```
../src/server/scripts/Commands/cs_event.cpp:104:48: error: no matching function for call to ‘std::set<short unsigned int>::find(const Trinity::ChatCommands::Variant<Trinity::ChatCommands::Hyperlink<Trinity::Hyperlinks::LinkTags::gameevent>, unsigned int>&) const’
```

File: src/server/scripts/Commands/cs_event.cpp (hunk 2 above)
Comment (Treeston) [reply src/server/scripts/Commands/cs_event.cpp:88]: what is the tag base for the `gameevent` link tag? can you `static_cast` it to `uint32`?

**Review by Treeston — APPROVED** [review]

**Pull request merged**